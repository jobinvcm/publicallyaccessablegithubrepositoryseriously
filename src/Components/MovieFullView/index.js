import React from "react";
import Paper from "@material-ui/core/Paper";
import propTypes from 'prop-types';
import ImageFetch from '../../Services/imageLoad';
import { withStyles } from '@material-ui/core/styles';
import { Hidden } from "@material-ui/core";

const styles = theme => ({
    root: {
        position: "fixed",
        top: 0,
        left: 0,
        width: "100vw",
        height: "100vh",
        overflowY: "scroll",
        backgroundColor: "#081B23",
    },
    imageContainer: {
        maxWidth: "185px",
        overflow: "hidden",
        position: "relative",
        top: "-100px"
    },
    backgroundImageContainer: {
        width: "100%",
        maxHeight: "250px",
        [theme.breakpoints.up('md')] : {
            minHeight: "100vh"
        },
        overflow: "hidden",
        position: "relative",
    },
    backgroundImage: {
        minWidth: "100%",
        minHeight: "100%",
        display: "block"
    },
    profileImage: {
        display: "block",
        minWidth: "100%",
        minHeight: "100%",
    }
});

const MovieFullView = (props) => {
  const { classes } = props;
  let imageUrl = `https://image.tmdb.org/t/p/w185/${props.item.poster_path}`;
  
  return (
    <div className={classes.root}>
        <div className={classes.backgroundImageContainer}>
            {<ImageFetch style={classes.backgroundImage} imageUrl={props.item.backdrop_path}/>}
        </div>
      <Paper className={classes.imageContainer} elevation={2}>
        <img className={classes.profileImage} src={imageUrl} />
      </Paper>

      <div>{props.item.title}</div>
      <div>{props.item.year}</div>
      <div>{props.item.overview}</div>
    </div>
  );
};

MovieFullView.propTypes = {
  item: propTypes.shape({
    imageUrl: propTypes.string,
    title: propTypes.string,
    year: propTypes.string
  })
};

export default withStyles(styles)(MovieFullView);
