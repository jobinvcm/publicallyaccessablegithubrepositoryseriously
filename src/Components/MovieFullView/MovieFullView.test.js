import React from 'react';
import { shallow } from 'enzyme';
import MovieFullView from '.';
import Paper from '@material-ui/core/Paper';
import SearchIcon from "@material-ui/icons/Search";
import IconButton from "@material-ui/core/IconButton";
import InputBase from "@material-ui/core/InputBase";

describe('<MovieFullView />', () => {
  it('renders paper component', () => {
    const wrapper = shallow(<MovieFullView />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have query method', () => {
      const wrapper = shallow(<MovieFullView />);
      const movieView = wrapper.instance();
      console.log(movieView.buildQuery);
  })
});
