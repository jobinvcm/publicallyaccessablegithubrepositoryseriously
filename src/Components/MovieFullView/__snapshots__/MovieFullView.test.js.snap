// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieFullView /> renders paper component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(MovieFullView) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "backgroundImage": "MovieFullView-backgroundImage-4",
        "backgroundImageContainer": "MovieFullView-backgroundImageContainer-3",
        "details": "MovieFullView-details-17",
        "goBackIcon": "MovieFullView-goBackIcon-7",
        "goBackLink": "MovieFullView-goBackLink-8",
        "infoDetailsSection": "MovieFullView-infoDetailsSection-11",
        "infoImageSection": "MovieFullView-infoImageSection-10",
        "infoSection": "MovieFullView-infoSection-9",
        "itemDetails": "MovieFullView-itemDetails-13",
        "itemTitle": "MovieFullView-itemTitle-12",
        "notFound": "MovieFullView-notFound-6",
        "overviewContainer": "MovieFullView-overviewContainer-15",
        "overviewDetails": "MovieFullView-overviewDetails-16",
        "overviewTitle": "MovieFullView-overviewTitle-14",
        "profileImage": "MovieFullView-profileImage-5",
        "profileImageContainer": "MovieFullView-profileImageContainer-2",
        "root": "MovieFullView-root-1",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "backgroundImage": "MovieFullView-backgroundImage-4",
          "backgroundImageContainer": "MovieFullView-backgroundImageContainer-3",
          "details": "MovieFullView-details-17",
          "goBackIcon": "MovieFullView-goBackIcon-7",
          "goBackLink": "MovieFullView-goBackLink-8",
          "infoDetailsSection": "MovieFullView-infoDetailsSection-11",
          "infoImageSection": "MovieFullView-infoImageSection-10",
          "infoSection": "MovieFullView-infoSection-9",
          "itemDetails": "MovieFullView-itemDetails-13",
          "itemTitle": "MovieFullView-itemTitle-12",
          "notFound": "MovieFullView-notFound-6",
          "overviewContainer": "MovieFullView-overviewContainer-15",
          "overviewDetails": "MovieFullView-overviewDetails-16",
          "overviewTitle": "MovieFullView-overviewTitle-14",
          "profileImage": "MovieFullView-profileImage-5",
          "profileImageContainer": "MovieFullView-profileImageContainer-2",
          "root": "MovieFullView-root-1",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
